kolibri/plugins/coach/assets/src/views/CoachImmersivePage.vue

```diff
@@ -12,7 +12,12 @@
       :loading="loading"
       :appearanceOverrides="appearanceOverrides"
     >
-      <div class="coach-main">
+      <KCircularLoader
+        v-if="coreLoading"
```

Comment:
Some time ago, we attempted to move away from having global loading state being managed by these root pages but instead let it be managed by the child pages as such architecture showed up not being very flexible and on some cases preventing us from updating pages in a reasonable manner. I'm concerned that by introducing this again here will lead to same issues. I didn't dig into detail into all pages at this point, but here's one example where I see potential confusion https://github.com/learningequality/kolibri/blob/release-v0.16.x/kolibri/plugins/coach/assets/src/views/plan/QuizEditDetailsPage.vue#L93-L96

I didn't review in detail yet and can think about it more. Just looking at it very high-level, I think that this type of change may easily end up having unexpected impact and seems to go against the new pattern we've tried to adopt in the whole codebase to address problems by this approach. For now, I wanted to ask what was the reason for introducing it in this manner and if you've already considered alternatives.

Reply:
I know it'd be lots of work to do it from child pages. On the other hand, it seems that we'd need to have a look at them anyways to cleanup conflicting or overlapping loading states so that we don't run again into issues we've been cleaning up lately after the core base refactor.

Reply:
I did this because when the refactor happened to remove the global loading state from CoreBase, it wasn't then re-added to each individual page in coach, which for many of the pages were reliant on the global loading state preventing their components from being shown until state has been set.

As a result, all of these components make unguarded reads from their module state - or are not set up to reactively update if the module state changes while they are displayed. This is the cause of the bug that Peter has reported here.

I agree that we _should_ be moving in the direction you describe, but this is a bandaid to account for the incomplete refactor, and I think not doing it is far more likely to leave additional regressions undiscovered until after release.

The code you are pointing to has not changed for 4 years (long predating the refactor) so I think the chance of this change causing issues for the loading state there is minimal.

Reply:
I definitely agree it's caused by incomplete refactor and no problem to keep addressing this outside of the scope of this particular PR. My only suggestion would be to have an issue explaining why this change happened and in what direction we want to move on.

> The code you are pointing to has not changed for 4 years (long predating the refactor) so I think the chance of this change causing issues for the loading state there is minimal.

I would rather say that we don't know, but one thing we can do to be more sure is to test loading states thoroughly.

One last high-level question, before I look into details, why is circular loader used rather than the linear one?

Reply:
The only PRs that I am aware of that have made changes to the loading state since the refactor are this one: https://github.com/learningequality/kolibri/pull/11218 and this one: https://github.com/learningequality/kolibri/pull/11201

Neither of which affect pages that use the CoachImmersivePage, so I think the chance of a conflict between this change and previous loading state work. Unless I've missed a PR?

Reply:
> One last high-level question, before I look into details, why is circular loader used rather than the linear one?

The linear loader's appearance is somewhat subtle, and can easily not be noticed (at least by me) - plus there is already a linear loader in the immersive and app bar pages I believe, so adding another one below seemed a bit odd.

The alternative is not to include a loader here at all and just gate the main HTML tag with `v-if="!loading"`

Reply:
> Neither of which affect pages that use the CoachImmersivePage, so I think the chance of a conflict between this change and previous loading state work. Unless I've missed a PR?

It's possible, yes. I don't recall all PRs as I only worked on a small portion. @nucleogenesis and @marcellamaki may know more. From my perspective, in many places in Coach we moved towards handling loading state locally overall, that's why I think that introducing new global condition to one of the root pages would deserve some testing across Coach rather than just verifying the bug. Hopefully, it will be okay. It's just my recommendation based on various updates around loaders during the past few months.

> The linear loader's appearance is somewhat subtle, and can easily not be noticed (at least by me) - plus there is already a linear loader in the immersive and app bar pages I believe, so adding another one below seemed a bit odd.

> The alternative is not to include a loader here at all and just gate the main HTML tag with v-if="!loading"

Regarding technical side of things, I don't know what's better, taking into account that loading states are overall work in progress. So I think there's not a perfect solution at this point and either is okay.

Regarding UX and the subtlety of the linear loader, that's true. I usually recommend to follow [our current guidelines](https://develop--kolibri-design-system.netlify.app/loaders/#overview) to not complicate things and to not introduce inconsistencies into the app. I also see this as something to be discussed with designers at first, and then refactored in the whole Kolibri if we decide to move away from linear loaders. To prevent back and forth and to be on the same page with them and also between us, devs, rather then each of us approaching it differently. So from this point of view `v-if="!loading"` would make sense?

Reply:
Yes, that should work - I'll double check the linear loader is appearing properly for the immersive pages and remove the circular loader.

Reply:
Updated.

Reply:
I can also raise the issue of global vs linear loaders in a KDS tactical if that'd be helpful. I head more devs complaining about it :)

Reply:
I am not terribly fussed about doing away with global loaders (although I think having the linear loader consistently be turned on until route navigation has completed at least would be fairly easy to do globally, without having to mess around with anything in each route) - but getting rid of them completely would mean that we need to rearchitect all our pages to handle their loading states internally (which is probably best achieved by moving their loading state into `setup` in the component).

Reply:
> although I think having the linear loader consistently be turned on until route navigation has completed at least would be fairly easy to do globally, without having to mess around with anything in each route

Yes, I see value in this. Can't think of any issues in this particular case. I assume it should be pretty brief (after we deal with issues like https://github.com/learningequality/kolibri/issues/11219) so in that case it may even make more sense to stay with the subtler global linear loader.

Reply:
By "it" I meant route navigation, if we stop doing API requests there